### PR TITLE
[#3172] Remove hono-service-base-quarkus module

### PR DESCRIPTION
**Pull request by calohmn**
Second part of #3172.

### CODE REVIEW

**Comment by sophokles73** [comment]
@calohmn have you seen the failing CI build?

**Comment by calohmn** [comment]
The failing unit tests had this kind of error:
```
java.lang.IllegalStateException: SRCFG00017: Configuration already registered for the given class loader
	at io.smallrye.config.SmallRyeConfigProviderResolver.registerConfig(SmallRyeConfigProviderResolver.java:118)
	at io.quarkus.test.junit.BasicLoggingEnabler.beforeAll(BasicLoggingEnabler.java:102)
	at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.lambda$invokeBeforeAllCallbacks$10(ClassBasedTestDescriptor.java:381)
```
It seems to me that the `BasicLoggingEnabler` (added in https://github.com/quarkusio/quarkus/issues/7696) has issues with tests being run in parallel.
I'll look into creating a Quarkus issue here.

<details>
<summary>Verbose log output of failing 'hono-service-device-registry-base' test</summary>

~~~
[INFO] Running org.eclipse.hono.service.management.tenant.TrustedCertificateAuthorityTest
[INFO] Running org.eclipse.hono.deviceregistry.service.credentials.AbstractCredentialsServiceTest
[INFO] Running org.eclipse.hono.deviceregistry.service.device.AbstractDeviceManagementServiceTest
[INFO] Running org.eclipse.hono.deviceregistry.service.credentials.AbstractCredentialsManagementServiceTest
BasicLoggingEnabler took 137 ms from static init to start of beforeAll()
BasicLoggingEnabler took 138 ms from static init to start of beforeAll()
BasicLoggingEnabler took 138 ms from static init to start of beforeAll()
BasicLoggingEnabler took 138 ms from static init to start of beforeAll()
[INFO] Running org.eclipse.hono.deviceregistry.service.device.AbstractRegistrationServiceTest
BasicLoggingEnabler took 209 ms from static init to start of beforeAll()
[INFO] Running org.eclipse.hono.deviceregistry.service.device.EdgeDeviceAutoProvisionerTest
BasicLoggingEnabler took 210 ms from static init to start of beforeAll()
[INFO] Running org.eclipse.hono.deviceregistry.service.tenant.AbstractTenantManagementServiceTest
BasicLoggingEnabler took 220 ms from static init to start of beforeAll()
[INFO] Running org.eclipse.hono.deviceregistry.service.tenant.DefaultTenantInformationServiceTest
BasicLoggingEnabler took 263 ms from static init to start of beforeAll()
[INFO] Running org.eclipse.hono.deviceregistry.util.DeviceRegistryUtilsTest
BasicLoggingEnabler took 298 ms from static init to start of beforeAll()
[INFO] Running org.eclipse.hono.service.management.FilterTest
[INFO] Running org.eclipse.hono.service.management.SortTest
BasicLoggingEnabler took 343 ms from static init to start of beforeAll()
BasicLoggingEnabler took 346 ms from static init to start of beforeAll()
[INFO] Running org.eclipse.hono.service.management.credentials.CommonSecretTest
BasicLoggingEnabler took 359 ms from static init to start of beforeAll()
[INFO] Running org.eclipse.hono.service.management.credentials.CredentialsDtoTest
BasicLoggingEnabler took 374 ms from static init to start of beforeAll()
[INFO] Running org.eclipse.hono.service.management.credentials.CredentialsTest
BasicLoggingEnabler took 398 ms from static init to start of beforeAll()
[INFO] Running org.eclipse.hono.service.management.credentials.GenericSecretTest
BasicLoggingEnabler took 437 ms from static init to start of beforeAll()
[INFO] Running org.eclipse.hono.service.management.credentials.PasswordSecretTest
BasicLoggingEnabler took 448 ms from static init to start of beforeAll()
[INFO] Running org.eclipse.hono.service.management.credentials.PskSecretTest
BasicLoggingEnabler took 470 ms from static init to start of beforeAll()
[INFO] Running org.eclipse.hono.service.management.credentials.X509CertificateSecretTest
BasicLoggingEnabler took 476 ms from static init to start of beforeAll()
[INFO] Running org.eclipse.hono.service.management.device.DelegatingDeviceManagementHttpEndpointTest
BasicLoggingEnabler took 497 ms from static init to start of beforeAll()
[INFO] Running org.eclipse.hono.service.management.device.DeviceAndGatewayAutoProvisionerTest
BasicLoggingEnabler took 502 ms from static init to start of beforeAll()
[INFO] Running org.eclipse.hono.service.management.device.DeviceTest
BasicLoggingEnabler took 541 ms from static init to start of beforeAll()
BasicLoggingEnabler took 554 ms to load config
[INFO] Running org.eclipse.hono.service.management.tenant.TenantTest
BasicLoggingEnabler took 566 ms from static init to start of beforeAll()
BasicLoggingEnabler ConfigSource: SysPropConfigSource
BasicLoggingEnabler ConfigSource: EnvConfigSource
BasicLoggingEnabler ConfigSource: PropertiesConfigSource[source=jar:file:///home/cal/.m2/repository/org/eclipse/hono/hono-service-base/2.0.0-SNAPSHOT/hono-service-base-2.0.0-SNAPSHOT.jar!/application.properties]
BasicLoggingEnabler ConfigSource: io.quarkus.test.common.http.TestHTTPConfigSourceProvider$TestURLConfigSource@2397d752
BasicLoggingEnabler ConfigSource: PropertiesConfigSource[source=Runtime Defaults]
BasicLoggingEnabler ConfigSource: DefaultValuesConfigSource
BasicLoggingEnabler is swapping config after java.lang.IllegalStateException: SRCFG00017: Configuration already registered for the given class loader
BasicLoggingEnabler is swapping config after java.lang.IllegalStateException: SRCFG00017: Configuration already registered for the given class loader
BasicLoggingEnabler is swapping config after java.lang.IllegalStateException: SRCFG00017: Configuration already registered for the given class loader
BasicLoggingEnabler is swapping config after java.lang.IllegalStateException: SRCFG00017: Configuration already registered for the given class loader
BasicLoggingEnabler is swapping config after java.lang.IllegalStateException: SRCFG00017: Configuration already registered for the given class loader
BasicLoggingEnabler is swapping config after java.lang.IllegalStateException: SRCFG00017: Configuration already registered for the given class loader
BasicLoggingEnabler is swapping config after java.lang.IllegalStateException: SRCFG00017: Configuration already registered for the given class loader
BasicLoggingEnabler is swapping config after java.lang.IllegalStateException: SRCFG00017: Configuration already registered for the given class loader
BasicLoggingEnabler is swapping config after java.lang.IllegalStateException: SRCFG00017: Configuration already registered for the given class loader
BasicLoggingEnabler is swapping config after java.lang.IllegalStateException: SRCFG00017: Configuration already registered for the given class loader
BasicLoggingEnabler is swapping config after java.lang.IllegalStateException: SRCFG00017: Configuration already registered for the given class loader
BasicLoggingEnabler is swapping config after java.lang.IllegalStateException: SRCFG00017: Configuration already registered for the given class loader
BasicLoggingEnabler is swapping config after java.lang.IllegalStateException: SRCFG00017: Configuration already registered for the given class loader
BasicLoggingEnabler is swapping config after java.lang.IllegalStateException: SRCFG00017: Configuration already registered for the given class loader
BasicLoggingEnabler is swapping config after java.lang.IllegalStateException: SRCFG00017: Configuration already registered for the given class loader
BasicLoggingEnabler is swapping config after java.lang.IllegalStateException: SRCFG00017: Configuration already registered for the given class loader
BasicLoggingEnabler is swapping config after java.lang.IllegalStateException: SRCFG00017: Configuration already registered for the given class loader
BasicLoggingEnabler is swapping config after java.lang.IllegalStateException: SRCFG00017: Configuration already registered for the given class loader
BasicLoggingEnabler is swapping config after java.lang.IllegalStateException: SRCFG00017: Configuration already registered for the given class loader
BasicLoggingEnabler is swapping config after java.lang.IllegalStateException: SRCFG00017: Configuration already registered for the given class loader
BasicLoggingEnabler is swapping config after java.lang.IllegalStateException: SRCFG00017: Configuration already registered for the given class loader
[ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 0.132 s <<< FAILURE! - in org.eclipse.hono.deviceregistry.service.credentials.AbstractCredentialsServiceTest
[ERROR] org.eclipse.hono.deviceregistry.service.credentials.AbstractCredentialsServiceTest  Time elapsed: 0.13 s  <<< ERROR!
java.lang.IllegalStateException: SRCFG00017: Configuration already registered for the given class loader
	at io.smallrye.config.SmallRyeConfigProviderResolver.registerConfig(SmallRyeConfigProviderResolver.java:118)
	at io.quarkus.test.junit.BasicLoggingEnabler.beforeAll(BasicLoggingEnabler.java:102)
	at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.lambda$invokeBeforeAllCallbacks$10(ClassBasedTestDescriptor.java:381)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.invokeBeforeAllCallbacks(ClassBasedTestDescriptor.java:381)
	at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.before(ClassBasedTestDescriptor.java:205)
	at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.before(ClassBasedTestDescriptor.java:80)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:148)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:141)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:139)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:138)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:95)
	at org.junit.platform.engine.support.hierarchical.ForkJoinPoolHierarchicalTestExecutorService$ExclusiveTask.compute(ForkJoinPoolHierarchicalTestExecutorService.java:185)
	at java.base/java.util.concurrent.RecursiveAction.exec(RecursiveAction.java:189)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:290)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1020)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1656)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1594)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:183)

BasicLoggingEnabler took 443 ms from start of beforeAll() to end
BasicLoggingEnabler took 315 ms from start of beforeAll() to end
BasicLoggingEnabler took 744 ms from start of beforeAll() to end
BasicLoggingEnabler took 522 ms from start of beforeAll() to end
BasicLoggingEnabler took 433 ms from start of beforeAll() to end
BasicLoggingEnabler took 671 ms from start of beforeAll() to end
BasicLoggingEnabler took 406 ms from start of beforeAll() to end
BasicLoggingEnabler took 342 ms from start of beforeAll() to end
BasicLoggingEnabler took 537 ms from start of beforeAll() to end
BasicLoggingEnabler took 410 ms from start of beforeAll() to end
BasicLoggingEnabler took 383 ms from start of beforeAll() to end
BasicLoggingEnabler took 660 ms from start of beforeAll() to end
BasicLoggingEnabler took 742 ms from start of beforeAll() to end
BasicLoggingEnabler took 482 ms from start of beforeAll() to end
BasicLoggingEnabler took 742 ms from start of beforeAll() to end
BasicLoggingEnabler took 506 ms from start of beforeAll() to end
BasicLoggingEnabler took 554 ms from start of beforeAll() to end
BasicLoggingEnabler took 621 ms from start of beforeAll() to end
BasicLoggingEnabler took 675 ms from start of beforeAll() to end
BasicLoggingEnabler took 583 ms from start of beforeAll() to end
BasicLoggingEnabler took 379 ms from start of beforeAll() to end
~~~

</details>

In this PR, I've disabled parallel test execution for the modules where this kind of error happened.

**Comment by sophokles73** [comment]
> It seems to me that the BasicLoggingEnabler (added in https://github.com/quarkusio/quarkus/issues/7696) has issues with tests being run in parallel.

If that is the case, then why didn't this occur before, or did it? I assume we are using this configuration in other modules as well, don't we?

**Comment by sophokles73** [comment]
@calohmn when I run the unit tests in paralle, that you have disabled parallel execution for, then everything works fine. The only problem I see is that there are multiple implementations of SLF4J on the class path because you have removed the corresponding exclusions from the service-base's quarkus-core dependency ...

**Comment by sophokles73** [comment]
@calohmn a good option should also be to simply selectively exclude the *slf4j-jboss-logmanager* dependency when running the unit tests by means of configuring the maven surefire plugin accordingly (see https://github.com/eclipse/hono/commit/853addcf104d904b3685be0a1fd6c058d4fc9eed)

**Comment by calohmn** [comment]
> My understanding is that Quarkus does not (yet) support running standard unit tests using the JBoss logging framework being configured form the application.yaml.

To me it looks like this support has been added as part of the aforementioned https://github.com/quarkusio/quarkus/issues/7696. This can be verified with the `hono-client-kafka-common` module.
Editing 
`clients/kafka-common/src/test/resources/application.yaml`
and adding
```
quarkus:
  log:
    console:
      color: true
    level: INFO
    min-level: TRACE
    category:
      "org.eclipse.hono":
        level: DEBUG
      "org.apache.kafka":
        level: DEBUG
```
there, means that running tests like `KafkaClientFactoryTest` in that module will produce proper DEBUG log output.


> IMHO we should use logback for running unit tests and (where possible) exclude the slf4j-jboss-logmanager dependency of quarkus-core in modules that contain unit tests.

> a good option should also be to simply selectively exclude the *slf4j-jboss-logmanager* dependency when running the unit tests by means of configuring the maven surefire plugin accordingly (see https://github.com/eclipse/hono/commit/853addcf104d904b3685be0a1fd6c058d4fc9eed)

I also think that this `slf4j-jboss-logmanager` exclusion can only be done via this Maven Surefire plugin `classpathDependencyExclude` then.
But, when running tests via the IDE instead of via Maven, that kind of exclusion isn't getting applied. So, when developing and running tests from the IDE, there is no proper log output.
Therefore, I would prefer not to go that way.

> The fact that we mostly have unit tests in the base modules allows for explicitly adding the slf4j-jboss-logmanager dependency in the Quarkus specific implementation modules.

The point is that with #3172 and this PR, this separation of Quarkus-independent base modules and Quarkus-specific implementation modules is being removed, adding the Quarkus dependencies already in the base modules.

That all means that I think we should either migrate towards not using logback any more, or skip/undo the changes for #3172 entirely.
The disadvantage of the former is no parallel test execution in corresponding module, the disadvantage of the latter is having a more complex module structure with the extra quarkus-base modules.

**Comment by sophokles73** [comment]
This [comment](https://github.com/quarkusio/quarkus/blob/acf5b28db1825edc5483fc2472c80ca965c553ce/test-framework/junit5/src/main/java/io/quarkus/test/junit/BasicLoggingEnabler.java#L117) seems to suggest that parallel execution might indeed be problematic. If there already is a potential race condition when running test serially, I am pretty sure that there is a race condition when running in parallel ...

**Comment by sophokles73** [comment]
> But, when running tests via the IDE instead of via Maven, that kind of exclusion isn't getting applied. So, when developing and running tests from the IDE, there is no proper log output.

Good point

**Comment by calohmn** [comment]
(rebased)

**Comment by calohmn** [comment]
Created https://github.com/quarkusio/quarkus/issues/24524 regarding the test failures when run in parallel.

File: tests/pom.xml [hunk 1]
```diff
@@ -231,6 +231,12 @@
       <groupId>org.eclipse.hono</groupId>
       <artifactId>hono-service-device-registry-base</artifactId>
       <scope>test</scope>
+      <exclusions>
+        <exclusion>
+          <groupId>org.jboss.slf4j</groupId>
+          <artifactId>slf4j-jboss-logmanager</artifactId>
+        </exclusion>
+      </exclusions>
```
Comment (sophokles73) [inline tests/pom.xml:239]: FMPOV we should keep either this exclusion or the configuration of the failsafe plugin, but not both

Comment (calohmn) [reply tests/pom.xml:239]: This must have come in via the extra "Exclude JBoss logging slf4j binding when running ITs" commit in the list. I didn't knowingly add it there. I'll rebase...

Comment (calohmn) [reply tests/pom.xml]: Change is not in this PR any more now.

**Comment by calohmn** [comment]
Regarding the `logback-test.xml` configurations: There is also the https://github.com/quarkiverse/quarkus-logging-logback extension that adds logback support. But this requires Quarkus to get initialized, so it's only working with `@QuarkusTest` tests.

**Review by sophokles73 — APPROVED** [review]
LGTM

**Comment by calohmn** [comment]
It looks like the issue with the parallel test execution will be fixed with the next Quarkus update. The 2nd commit here can be reverted then. 
In general, usage of the `quarkus-junit5` dependency and change from logback to quarkus logging config in all adapter and service modules is to be done in a follow-up commit/PR.

**Pull request merged**